crates/evm/src/tests/ef_tests/cases/blockchain_test.rs

```diff
@@ -0,0 +1,291 @@
+                                assert_eq!(U256::from(account_state.info.nonce), account.nonce);
+                                assert_eq!(account_state.info.balance, account.balance);
+                            }
+                        }
+                    }
+                    (None, Some(expected_state_root)) => {
+                        // Insert state hashes into the provider based on the expected state root.
+                        assert_eq!(
+                            *evm.head.get(&mut working_set).unwrap().header.state_root,
+                            **expected_state_root
+                        );
+                    }
+                    _ => return Err(Error::MissingPostState),
+                }
+
+                Ok(())
+            })?;
+
+        Ok(())
+    }
+}
+
+/// Returns whether the test at the given path should be skipped.
+///
+/// Some tests are edge cases that cannot happen on mainnet, while others are skipped for
+/// convenience (e.g. they take a long time to run) or are temporarily disabled.
+///
+/// The reason should be documented in a comment above the file name(s).
+pub fn should_skip(path: &Path) -> bool {
+    let path_str = path.to_str().expect("Path is not valid UTF-8");
+    let name = path.file_name().unwrap().to_str().unwrap();
+    matches!(
+        name,
+        // funky test with `bigint 0x00` value in json :) not possible to happen on mainnet and require
+        // custom json parser. https://github.com/ethereum/tests/issues/971
+        | "ValueOverflow.json"
+        | "ValueOverflowParis.json"
+
+        // txbyte is of type 02 and we dont parse tx bytes for this test to fail.
+        | "typeTwoBerlin.json"
+
+        // Test checks if nonce overflows. We are handling this correctly but we are not parsing
+        // exception in testsuite There are more nonce overflow tests that are in internal
+        // call/create, and those tests are passing and are enabled.
+        | "CreateTransactionHighNonce.json"
+
+        // Test check if gas price overflows, we handle this correctly but does not match tests specific
+        // exception.
+        | "HighGasPrice.json"
+        | "HighGasPriceParis.json"
+
+        // Skip test where basefee/accesslist/difficulty is present but it shouldn't be supported in
+        // London/Berlin/TheMerge. https://github.com/ethereum/tests/blob/5b7e1ab3ffaf026d99d20b17bb30f533a2c80c8b/GeneralStateTests/stExample/eip1559.json#L130
+        // It is expected to not execute these tests.
+        | "accessListExample.json"
+        | "basefeeExample.json"
+        | "eip1559.json"
+        | "mergeTest.json"
+
+        // These tests are passing, but they take a lot of time to execute so we are going to skip them.
+        | "loopExp.json"
+        | "Call50000_sha256.json"
+        | "static_Call50000_sha256.json"
+        | "loopMul.json"
+        | "CALLBlake2f_MaxRounds.json"
+        | "shiftCombinations.json"
+    )
```

Comment:
Should we include them in the daily running hive tests? As a job that executes once in every day so that we notice if anything goes wrong?

Reply:
those are not running

Reply:
included in the other ef-tests issue now